tests/test_ble_lifecycle_controller.py

```
@@ -0,0 +1,555 @@
+
+        configured_hook.assert_called_once_with()
+        assert resolved_is_closing == [True]
+
+    def test_resolve_hook_with_unconfigured_mock_returns_fallback(self) -> None:
+        """_resolve_hook should return fallback when hook is an unconfigured mock."""
+        # An unconfigured MagicMock (default behavior)
+        unconfigured_hook = MagicMock()
+
+        mock_iface = SimpleNamespace(_state_manager_is_closing=unconfigured_hook)
+        mock_ownership = SimpleNamespace(
+            _discard_invalidated_connected_client=lambda *a, **kw: None
```

Ruff (0.15.6)

[warning] Unused lambda argument: `a`

(ARG005)